rtc2git.cli.extension/src/to/rtc/cli/migrate/HistoryEntryVisitor.java

```diff
@@ -56,14 +81,96 @@ protected void visitChangeSet(ChangeLogEntryDTO parent, ChangeLogChangeSetEntryD
+		final boolean wasAdded = actualTag.add(changeSet);
+		debugLog(2,(wasAdded?"Added":"FAILED to add") + " changeSet " + changeSetUuid + " to tag " + actualTag);
 		if (lastChangeSets.get(component).equals(changeSetUuid)) {
 			lastChangeSetReached = true;
 			actualTag.setContainLastChangeset(true);
+			debugLog(2,"This was the last changeset in tag "+actualTag);
 		}
 	}
 
+	@Override
+	protected void visitBaseline(ChangeLogEntryDTO parent, ChangeLogBaselineEntryDTO dto) {
+		debugLog("HEV.visitBaseline(" + dtoToString(parent) +", " + dtoToString(dto) + ")");
+		debugLog(2,"BaselineId="+dto.getBaselineId());
+		debugLog(2,"CreationDate="+RtcChangeSet.getDateAsString(dto.getCreationDate()));
+		debugLog(2,"Creator.FullName="+(dto.getCreator()!=null?dto.getCreator().getFullName():null));
+		debugLog(2,"EntryName="+dto.getEntryName());
+		debugLog(2,"EntryType="+dto.getEntryType());
+		super.visitBaseline(parent, dto);
+	}
+
+	//@Override
+	//protected void visitDirection(ChangeLogEntryDTO parent, ChangeLogDirectionEntryDTO dto) {
+	//	debugLog("HEV.visitDirection(" + dtoToString(parent) +", " + dtoToString(dto) + ")");
+	//	debugLog(2,"EntryName="+dto.getEntryName());
+	//	debugLog(2,"EntryType="+dto.getEntryType());
+	//	debugLog(2,"FlowDirection="+dto.getFlowDirection());
+	//	super.visitDirection(parent, dto);
+	//}
+
+	//@Override
+	//protected void visitOslcLink(ChangeLogEntryDTO parent, ChangeLogOslcLinkEntryDTO dto, boolean inChangeSet) {
+	//	debugLog("HEV.visitOslcLink(" + dtoToString(parent) +", " + dtoToString(dto) + ", " + inChangeSet + ")");
+	//	super.visitOslcLink(parent, dto, inChangeSet);
+	//}
+
+	//@Override
+	//protected void visitVersionable(ChangeLogEntryDTO parent, ChangeLogVersionableEntryDTO dto) {
+	//	debugLog("HEV.visitVersionable(" + dtoToString(parent) +", " + dtoToString(dto) + ")");
+	//	debugLog(2,"EntryName="+dto.getEntryName());
+	//	debugLog(2,"EntryType="+dto.getEntryType());
+	//	debugLog(2,"Segments="+dto.getSegments());
+	//	super.visitVersionable(parent, dto);
+	//}
+
+	//@Override
+	//protected void visitWorkItem(ChangeLogEntryDTO parent, ChangeLogWorkItemEntryDTO dto, boolean inChangeSet) {
+	//	debugLog("HEV.visitWorkItem(" + dtoToString(parent) +", " + dtoToString(dto) + ", " + inChangeSet + ")");
+	//	debugLog(2,"EntryName="+dto.getEntryName());
+	//	debugLog(2,"EntryType="+dto.getEntryType());
+	//	debugLog(2,"Resolver.FullName="+(dto.getResolver()!=null?dto.getResolver().getFullName():null));
+	//	debugLog(2,"WorkItemNumber="+dto.getWorkItemNumber());
+	//	super.visitWorkItem(parent, dto, inChangeSet);
+	//}
+
+	//private int indent = 0;
+	//@Override
+	//public boolean enter(ChangeLogEntryDTO item) {
+	//	debugLog("HEV.enter(" + dtoToString(item) + ")");
+	//	indent++;
+	//	return super.enter(item);
+	//}
+
+	//@Override
+	//public void exit(ChangeLogEntryDTO item) {
+	//	indent--;
+	//	debugLog("HEV.exit(" + dtoToString(item) + ")");
+	//	super.exit(item);
+	//}
```

Comment:
I have mixed feelings about this.

This isn't code we'd want active 99% of the time; it hammers performance and spams the logs.
...but, when you're struggling to understand WTF is happening on the RTC side, this is essential.

So, in an ideal world, this code should be uncommented but made conditional, controlled by a config setting so that it can be turned on when it's really needed.
... or you can delete it and hope you don't regret doing so later.